core/src/main/java/feast/core/dao/FeatureSetRepository.java

```diff
@@ -21,26 +21,36 @@
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.Query;
 
-/** JPA repository supplying FeatureSet objects keyed by id. */
+/**
+ * JPA repository supplying FeatureSet objects keyed by id.
+ */
 public interface FeatureSetRepository extends JpaRepository<FeatureSet, String> {
 
   long count();
 
-  // Find feature set by name and version
-  FeatureSet findFeatureSetByNameAndVersion(String name, Integer version);
+  // Find single feature set by project, name, and version
+  FeatureSet findFeatureSetByNameAndProject_NameAndVersion(String name, String project,
+      Integer version);
+
+  // Find single latest version of a feature set by project and name
+  FeatureSet findFirstFeatureSetByNameAndProject_NameOrderByVersionDesc(String name,
+      String project);
+
+  // find all feature sets and order by name and version
+  List<FeatureSet> findAllByProject_ArchivedOrderByNameAscVersionAsc(boolean project_archived);
 
-  // Find latest version of a feature set by name
-  FeatureSet findFirstFeatureSetByNameOrderByVersionDesc(String name);
+  // find all feature sets within a project and order by name and version
+  List<FeatureSet> findAllByProject_NameAndProject_ArchivedOrderByNameAscVersionAsc(String project_name, boolean project_archived);
```

Comment:
Is it necessary to search by both project name and project archived? Will there ever be records with the same project name where there are records with archived=true and false?

Reply:
Project names are unique. So it wont happen. But I see value in this because I can list feature sets from archived projects as well as active projects

Do you have an alternative?

Reply:
If you're already searching by projects I dont really see a need to further filter by archived, since project name and project archived are 1:1

Reply:
Yes, but how do you prevent archived projects from being returned?

Reply:
Good point.